packages/medusa/src/api/routes/admin/discounts/delete-condition.ts

```diff
@@ -0,0 +1,79 @@
+import { IsOptional, IsString } from "class-validator"
+import { defaultAdminDiscountsFields, defaultAdminDiscountsRelations } from "."
+import { Discount } from "../../../../models"
+import { DiscountService } from "../../../../services"
+import DiscountConditionService from "../../../../services/discount-condition"
+import { getRetrieveConfig } from "../../../../utils/get-query-config"
+import { validator } from "../../../../utils/validator"
+
+/**
+ * @oas [delete] /discounts/{id}/conditions/{condition_id}
+ * operationId: "DeleteDiscountsDiscountCondition"
+ * summary: "Delete a DiscountCondition"
+ * description: "Deletes a DiscountCondition"
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The id of the Discount
+ *   - (path) condition_id=* {string} The id of the DiscountCondition
+ * tags:
+ *   - Discount
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             id:
+ *               type: string
+ *               description: The id of the deleted DiscountCondition
+ *             object:
+ *               type: string
+ *               description: The type of the object that was deleted.
+ *             deleted:
+ *               type: boolean
+ *             discount:
+ *               type: object
+ *               description: The Discount to which the condition used to belong
+ */
+export default async (req, res) => {
+  const { discount_id, condition_id } = req.params
+
+  const validatedParams = await validator(
+    AdminDeleteDiscountsConditionsConditionParams,
+    req.query
+  )
+
+  const conditionService: DiscountConditionService = req.scope.resolve(
+    "discountConditionService"
+  )
+  const discountService: DiscountService = req.scope.resolve("discountService")
+
+  await conditionService.remove(condition_id)
```

Comment:
before removing the condition here, we should retrieve the condition and inner joining the discount to ensure the condition exists on the actual discount from where the user asked to remove it.

Reply:
We'll have to do this in two steps; 1) retrieve discount to check if it exists, and 2) remove condition, because the inner join would otherwise fail on non-existing condition right? And we don't want to throw on non-existing entities but instead be idempotent @adrien2p. Do you have any suggestions?

Reply:
As it is at the moment, the retrieve method throws if there is no condition found. But the name retrieve does not imply that it can fail otherwise it should be named retrieveOrFail. So with how things works you have to retrieve the discount first and throw if it does not exists. That step is just a guard to ensure the following process is done in the right context. Then, you can do anything with the condition.